src/Symfony/Component/Security/Core/User/ChainUserProvider.php

```diff
@@ -104,4 +104,16 @@ public function supportsClass($class)
 
         return false;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        foreach ($this->providers as $provider) {
+            if ($provider instanceof PasswordUpgraderInterface) {
+                $provider->upgradePassword($user, $newEncodedPassword);
```

Comment:
This should catch `UnsupportedUserException` and try the next one when this happens. Otherwise if the user is coming from the second provider in the chain, but we ask the first one to upgrade the password first it might throw that exception and the whole thing would fail.

Ideally the interface would be extended with `canUpgradePasswordOf(UserInterface $u): bool` so you check upfront which provider is able to this for you instead of trying fist-first.

```
try {
   $provider->upgradePassword($user, $newEncodedPassword);
} catch (UnsupportedUserException) {
   // try the next one
}
```

Reply:
done thanks